app/models/ccms/submission.rb

```diff
@@ -21,11 +21,11 @@ def process!(options = {}) # rubocop:disable Metrics/MethodLength
       when 'applicant_submitted'
         CheckApplicantStatusService.call(self)
       when 'case_created'
-        ObtainDocumentIdService.call(self)
-      when 'document_ids_obtained'
         UploadDocumentsService.call(self)
-      when 'applicant_ref_obtained'
+      when 'document_ids_obtained'
         AddCaseService.call(self, options)
+      when 'applicant_ref_obtained'
+        ObtainDocumentIdService.call(self)
```

Comment:
I would suggest these should be in the logical order of the events firing, in order to make it a bit more obvious to the reader, both here and in the list of states in `CCMSSubmissionStateMachine`, so something like this:
```case aasm_state
      when 'initialised'
        ObtainCaseReferenceService.call(self)
      when 'case_ref_obtained'
        ObtainApplicantReferenceService.call(self)
      when 'applicant_submitted'
        CheckApplicantStatusService.call(self)
      when 'applicant_ref_obtained'
        ObtainDocumentIdService.call(self)
      when 'document_ids_obtained'
        AddCaseService.call(self, options)
      when 'case_submitted'
        CheckCaseStatusService.call(self)
      when 'case_created'
        UploadDocumentsService.call(self)
      else
        raise CcmsError, "Unknown state: #{aasm_state}"
      end```

Reply:
Makes sense. I've pushed a change doing that.